src/eth_provider/provider.rs

```diff
@@ -519,15 +519,12 @@ where
 
         // Update pending transactions collection
         self.database
-            .inner()
-            .collection::<StoredTransaction>("transactions_pending")
-            .update_one(
+            .update_one::<StoredPendingTransaction>(
                 doc! {"tx.hash": transaction_document.get_document("tx").unwrap().get_str("hash").unwrap()},
                 UpdateModifications::Document(doc! {"$set": transaction_document}),
                 UpdateOptions::builder().upsert(true).build(),
             )
```

Comment:
what do you think about also moving this section into the update one wrapper?

Reply:
I think that it can reduce the flexibility and generality of the update one function no? For example, for the set part (`UpdateModifications::Document(doc! {"$set": transaction_document})`) we could indeed put only `transaction_document` as an argument and abstract the rest into the wrapper function so that `UpdateModifications::Document(doc! {"$set": transaction_document})` is in the `update_one` function itself. But there is also an aggregation pipeline option in `UpdateModifications` and therefore we could not use it or we would have to do another function.

I know that the call here remains a bit heavy and the refactoring is not that obvious but to remain generic I took this option.

Perhaps what we could do would be to actually include these elements inside the function as you propose and if necessary in the future we can imagine other query or filter building structures to keep the generic aspect of our `update_one` function what do you think?

Reply:
> Perhaps what we could do would be to actually include these elements inside the function as you propose and if necessary in the future we can imagine other query or filter building structures to keep the generic aspect of our update_one function what do you think?

I think we can go that way, for now let's do this and we can think about a smarter way if we ever need to extend it.

Reply:
Just updated